plugins/code-coverage/src/api.ts

```diff
@@ -42,18 +45,26 @@ export const codeCoverageApiRef = createApiRef<CodeCoverageApi>({
 });
 
 export class CodeCoverageRestApi implements CodeCoverageApi {
-  url: string = '';
+  private readonly discoveryApi: DiscoveryApi;
+  private readonly fetchApi: FetchApi;
+
+  private url: string = '';
 
-  constructor(public discovery: DiscoveryApi) {}
+  public constructor(options: {
+    discoveryApi: DiscoveryApi;
+    fetchApi: FetchApi;
+  }) {
+    this.discoveryApi = options.discoveryApi;
+    this.fetchApi = options.fetchApi;
+  }
 
   private async fetch<T = unknown | string | JsonCoverageHistory>(
     path: string,
-    init?: RequestInit,
   ): Promise<T | string> {
     if (!this.url) {
-      this.url = await this.discovery.getBaseUrl('code-coverage');
+      this.url = await this.discoveryApi.getBaseUrl('code-coverage');
```

Comment:
Hey while we're here, might I trouble you for removing the `this.url` variable entirely and instead re-asking for the URL every single time? That's part of the contract for the discovery API, since it'd possibly return a new URL every single time in a round robin fashion or similar when scaling deployments horizontally.

Reply:
Sure, should be fixed now!